cmd/admin-handlers.go

```diff
@@ -473,3 +475,60 @@ func (adminAPI adminAPIHandlers) HealObjectHandler(w http.ResponseWriter, r *htt
 	// Return 200 on success.
 	writeSuccessResponseHeadersOnly(w)
 }
+
+// HealFormatHandler - POST /?heal
+// - x-minio-operation = format
+// - bucket and object are both mandatory query parameters
+// Heal a given object, if present.
+func (adminAPI adminAPIHandlers) HealFormatHandler(w http.ResponseWriter, r *http.Request) {
+	// Get current object layer instance.
+	objectAPI := newObjectLayerFn()
+	if objectAPI == nil {
+		writeErrorResponse(w, ErrServerNotInitialized, r.URL)
+		return
+	}
+
+	// Validate request signature.
+	adminAPIErr := checkRequestAuthType(r, "", "", "")
+	if adminAPIErr != ErrNone {
+		writeErrorResponse(w, adminAPIErr, r.URL)
+		return
+	}
+
+	// Create a new set of storage instances to heal format.json.
+	bootstrapDisks, err := initStorageDisks(globalEndpoints)
+	if err != nil {
+		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
+		return
+	}
+
+	// Heal format.json on available storage.
+	err = healFormatXL(bootstrapDisks)
+	if err != nil {
+		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
+		return
+	}
+
+	// Instantiate new object layer with newly formatted storage.
+	newObjectAPI, err := newXLObjects(bootstrapDisks)
+	if err != nil {
+		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
+		return
+	}
+
+	// Set object layer with newly formatted storage to globalObjectAPI.
+	globalObjLayerMutex.Lock()
+	{
+		globalObjectAPI = newObjectAPI
+	}
+	globalObjLayerMutex.Unlock()
+
+	// Shutdown storage belonging to old object layer instance.
```

Comment:
This looks neat that how it can work.. May you should mention the scoping done here..